src/Helpers/helperUploadPath.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+if (!function_exists('getUploadPath')) {
+
+    /**
+     * Return upload path.
+     *
+     * @param string $slug
+     * @param string $path
+     *
+     * @return string
+     */
+    function getUploadPath($slug, $path)
+    {
+        $_dirs = explode('/', $path);
+        $_out = '';
+
+        foreach ($_dirs as $dir) {
+            if ('' === $dir) {
+                continue;
+            }
+
+            if (preg_match('/%(.*?)%/', $dir)) {
+                $dir = str_replace('%', '', $dir);
+
+                if ($dir === 'slug') {
```

Comment:
I'm also not really a huge fan of the placeholder replacement.  This if/elseif/etc chain could easily get VERY long and unmanageable.